relay-redis/src/real.rs

```diff
@@ -397,143 +390,79 @@ pub struct Stats {
-            }
-            Self::Single(pool, ..) => {
-                let mut conn = pool.get().await.map_err(RedisError::AsyncPool)?;
-                cmd.query_async(&mut *conn).await.map_err(RedisError::Redis)
-            }
-        }
-    }
-
-    /// Creates a new cluster based [`AsyncRedisPool`].
-    ///
-    /// It will set the `response_timeout` to the provided `read_timeout`.
-    /// `write_timeout` is not used at all in the async pool
+impl AsyncRedisConnection {
+    /// Creates a [`AsyncRedisConnection`] in cluster mode.
     pub async fn cluster<'a>(
         servers: impl IntoIterator<Item = &'a str>,
         opts: &RedisConfigOptions,
     ) -> Result<Self, RedisError> {
         // connection timeout is set in `base_pool_builder` on the pool level
         let client = ClusterClientBuilder::new(servers)
             .response_timeout(Duration::from_secs(opts.read_timeout))
+            .connection_timeout(Duration::from_secs(opts.connection_timeout))
             .build()
             .map_err(RedisError::Redis)?;
-        let manager = RedisClusterConnectionManager::new(client);
-        let pool = Self::base_pool_builder(opts)
-            .build(manager)
+        let connection = client
+            .get_async_connection()
             .await
             .map_err(RedisError::Redis)?;
-        Ok(AsyncRedisPool::Cluster(pool, opts.clone()))
+        Ok(Self::Cluster(ClusterConnectionDebug(connection)))
     }
 
-    /// Creates a new [`AsyncRedisPool`] backed by a single redis server.
-    ///
-    /// It will set the `response_timeout` to the provided `read_timeout`.
-    /// `write_timeout` is not used at all in the async pool
+    /// Create a [`AsyncRedisConnection`] in single mode.
     pub async fn single(server: &str, opts: &RedisConfigOptions) -> Result<Self, RedisError> {
-        let manager = RedisConnectionManager::new(server).map_err(RedisError::Redis)?;
-        let customizer = Box::new(ResponseTimeoutCustomConnection(Duration::from_secs(
-            opts.read_timeout,
-        )));
-        // `connection_timeout` is set in `base_pool_builder` on the pool level
-        let pool = Self::base_pool_builder(opts)
-            .connection_customizer(customizer)
-            .build(manager)
+        let client = Client::open(server).map_err(RedisError::Redis)?;
+        let config = ConnectionManagerConfig::new()
+            .set_response_timeout(Duration::from_secs(opts.read_timeout))
+            .set_connection_timeout(Duration::from_secs(opts.connection_timeout));
+        let connection_manager = ConnectionManager::new_with_config(client, config)
             .await
             .map_err(RedisError::Redis)?;
-        Ok(AsyncRedisPool::Single(pool, opts.clone()))
+        Ok(Self::Single(ConnectionManagerDebug(connection_manager)))
     }
 
-    /// Returns information about the current state of the pool.
-    pub fn stats(&self) -> Stats {
```

Comment:
I think we can keep this and just return `connections: 1` and `idle: 0` until we have/need better stats.